ocs_ci/deployment/helpers/storage_class.py

```diff
@@ -0,0 +1,64 @@
+import logging
+import yaml
+
+from ocs_ci.framework import config
+from ocs_ci.framework.logger_helper import log_step
+from ocs_ci.ocs import constants
+from ocs_ci.utility.utils import run_cmd
+
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_STORAGE_CLASS_MAP = {
+    constants.AWS_PLATFORM: "gp2-csi",
+    constants.IBMCLOUD_PLATFORM: "ibmc-vpc-block-10iops-tier",
+    constants.VSPHERE_PLATFORM: "thin-csi",
+    constants.AZURE_PLATFORM: "managed-csi",
+    constants.GCP_PLATFORM: None,
+    constants.ROSA_HCP_PLATFORM: None,
+    constants.RHV_PLATFORM: "ovirt-csi-sc",
+}
+
+
+def get_storageclass() -> str:
+    """
+    Retrieve the storageclass to use from the config or based on platform
+
+    Returns:
+        str: Name of the storageclass
+
+    """
+    logger.info("Getting storageclass")
+    platform = config.ENV_DATA.get("platform")
+    customized_deployment_storage_class = config.DEPLOYMENT.get(
+        "customized_deployment_storage_class"
+    )
+
+    storage_class = DEFAULT_STORAGE_CLASS_MAP.get(platform)
```

Comment:
Wouldn't be better to use `DEFAULT_STORAGE_CLASS_MAP[platform]`, so it will directly raise a clear `KeyError:` exception in case we will implement new platform or is `None` a reasonable "default"?

In the case of raising `KeyError` exception for missing `platform`, it might be worth to combine this with the following condition for `customized_deployment_storage_class`, so in case the `platform` will be missing in the map, but `customized_deployment_storage_class` will be defined, it will not raise the exception.

Reply:
I originally had it this way but went with `.get` due to failures in [test_cloud.py](https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/deployment/tests/test_cloud.py). I could revert back to using a direct call to the key and change the test class to just set the proper attribute so it doesn't hit any issues.

For the `customized_deployment_storage_class` case I did add keys for the platforms which leverage this but I do think it's a good idea to check both before raising the error so I will incorporate that as well.

Reply:
I moved the check for `customized_deployment_storage_class` above setting the default value so we will check if that exists as a priority, and changed getting the default value to directly access the key.